astropy/visualization/units.py

```diff
@@ -19,13 +19,19 @@ def quantity_support(format="latex_inline"):
       >>> from astropy import units as u
       >>> from astropy import visualization
       >>> with visualization.quantity_support():
-      ...     plt.figure()
+      ...     fig = plt.figure()
       ...     plt.plot([1, 2, 3] * u.m)
```

Comment:
Hmm @neutrinoceros , didn't we move away from `plt` usage for some code in https://github.com/astropy/astropy/pull/17316 ? Should that be done where now that `fig` is used explicitly anyway?

Reply:
Hum, apparently I explicitly excluded `astropy/visualization` from my PR #17316, but I didn't state why. I'd *assume* I did it to keep the PR as small a possible, but I don't recall any reason *not* to do it in this subpackage too. However https://github.com/astropy/astropy/pull/17848 is still around, showing that it's not *always* trivial to conduct this migration everywhere, so here what I suggest we do here:
we can *try* avoiding `pyplot` in this test while we're at it (it should be easy™️, since outputs are ignored), but if that doesn't work for any reason, let's just leave it for future endeavors (maybe document what's not working, but don't spend too long digging into it now).